src/etos_lib/kubernetes/schemas/provider.py

```diff
@@ -0,0 +1,142 @@
+    stages: JSONTasIutPrepareStages
+
+
+class JSONTasIut(BaseProvider):
+    """JSONTasIUT describes the JSONTas specification of an IUT provider."""
+
+    prepare: Optional[JSONTasIutPrepare] = None
+
+
+class JSONTasExecutionSpace(BaseProvider):
+    """JSONTasExecutionSpace describes the JSONTas specification of an execution space provider."""
+
+
+class JSONTasLogArea(BaseProvider):
+    """JSONTasLogArea describes the JSONTas specification of a log area provider."""
+
+
+class JSONTas(BaseModel):
+    """JSONTas describes the JSONTas specification of a provider."""
+
+    iut: Optional[JSONTasIut] = None
+    execution_space: Optional[JSONTasExecutionSpace] = None
+    log: Optional[JSONTasLogArea] = None
+
+
+class Healthcheck(BaseModel):
+    """Healthcheck describes the healthcheck rules of a provider."""
+
+    endpoint: str
+    intervalSeconds: int
+
+
+class ProviderSpec(BaseModel):
+    """ProviderSpec is the specification of a Provider Kubernetes resource."""
+
+    type: str
+    host: Optional[str] = None
+    healthCheck: Optional[Healthcheck] = None
+    jsontas: Optional[JSONTas] = None
+
+
+class Provider(BaseModel):
+    """Provider Kubernetes resource."""
+
+    apiVersion: Optional[str] = "etos.eiffel-community.github.io/v1alpha1"
+    kind: Optional[str] = "Provider"
+    spec: ProviderSpec
+    metadata: Metadata
+
+    def to_jsontas(self) -> dict:
+        """To JSONTas will convert a provider to a JSONTas ruleset.
+
+        This method is for the transition period between the current ETOS and
+        the kubernetes controller based ETOS.
+        """
+        ruleset = {}
+        if self.spec.jsontas is not None:
+            if self.spec.type == "iut":
+                assert (
+                    self.spec.jsontas.iut is not None
+                ), "IUT must be a part of a Provider with type 'iut'."
+                ruleset = self.spec.jsontas.iut.model_dump()
+            elif self.spec.type == "execution-space":
+                assert (
+                    self.spec.jsontas.execution_space is not None
+                ), "Execution space must be a part of a Provider with type 'execution-space'."
+                ruleset = self.spec.jsontas.execution_space.model_dump()
+            elif self.spec.type == "log-area":
+                assert (
+                    self.spec.jsontas.log is not None
+                ), "Log area must be a part of a Provider with type 'log-area'."
+                ruleset = self.spec.jsontas.log.model_dump()
+        ruleset["id"] = self.metadata.name
+        return ruleset
+
+    def to_external(self) -> dict:
+        """To external will convert a provider to an external provider ruleset.
+
+        This method is for the transition period between the current ETOS and
+        the kubernetes controller based ETOS.
+        """
+        return {
+            "id": self.metadata.name,
+            "type": "external",
+            "status": {"host": f"{self.spec.host}/status"},
```

Comment:
self.spec.host is defined as optional. Is it correct?

Reply:
It is optional if JSONTas is set, otherwise it is not, making this correct.
This is checked in the ETOS controller so that the provders created in Kubernetes follow this rule